src/components/profile/Recovery/RecoveryContainer.js

```diff
@@ -0,0 +1,196 @@
+import { generateSeedPhrase } from 'near-seed-phrase';
+import {
+    setupRecoveryMessage,
+    deleteRecoveryMethod,
+    loadRecoveryMethods,
+    sendNewRecoveryLink
+} from '../../../actions/account';
+
+const Container = styled.div`
+
+    border: 2px solid #e6e6e6;
+    border-radius: 6px;
+
+    > div {
+        padding: 15px 20px;
+        border-bottom: 2px solid #f8f8f8;
+
+        &:last-of-type {
+            border-bottom: 0;
+        }
+    }
+
+    button {
+        font-size: 14px;
+        width: 100px;
+        font-weight: 600;
+        height: 40px;
+        letter-spacing: 0.5px;
+    }
+
+`
+
+const Header = styled.div`
+    padding: 20px !important;
+`
+
+const Title = styled.div`
+    font-family: BwSeidoRound;
+    color: #24272a;
+    font-size: 22px;
+    display: flex;
+    align-items: center;
+
+    &:before {
+        content: '';
+        background: center center no-repeat url(${RecoveryIcon});
+        width: 28px;
+        height: 28px;
+        display: inline-block;
+        margin-right: 10px;
+        margin-top: -2px;
+    }
+`
+
+const NoRecoveryMethod = styled.div`
+    margin-top: 15px;
+    color: #FF585D;
+    display: flex;
+    align-items: center;
+
+    &:before {
+        content: '';
+        background: center center no-repeat url(${ErrorIcon});
+        min-width: 28px;
+        width: 28px;
+        min-height: 28px;
+        height: 28px;
+        display: block;
+        margin-right: 10px;
+    }
+`
+
+class RecoveryContainer extends Component {
+
+    state = {
+        successSnackbar: false,
+        deletingMethod: '',
+        resendingLink: ''
+    };
+
+    handleEnableMethod = (method) => {
+        const { history, accountId } = this.props;
+
+        history.push(`${method !== 'phrase' ? '/set-recovery/' : '/setup-seed-phrase/'}${accountId}`);
+    }
+
+    handleDeleteMethod = (method) => {
+        const { deleteRecoveryMethod, loadRecoveryMethods, accountId } = this.props;
+
+        this.setState({ deletingMethod: method.kind })
+        deleteRecoveryMethod(method)
+            .then(({ error }) => {
+                if (error) return
+                loadRecoveryMethods(accountId);
+                this.setState({ deletingMethod: '' });
+        })
+    }
+
+    handleResendLink = (method) => {
+        const { seedPhrase, publicKey } = generateSeedPhrase();
+        const { accountId, sendNewRecoveryLink, loadRecoveryMethods } = this.props;
+        const { kind, detail } = method;
+        let phoneNumber, email;
+
+        if (kind === 'email') {
+            email = detail;
+        } else if (kind === 'phone') {
+            phoneNumber = detail;
+        }
+
+        this.setState({ resendingLink: method.kind })
+        sendNewRecoveryLink({ accountId, phoneNumber, email, publicKey, seedPhrase, method })
+            .then(({ error }) => {
+                if (error) return
+
+                loadRecoveryMethods(accountId);
```

Comment:
@vgrichina I `loadRecoveryMethods` here after re-sending link to get the latest data. However, it causes a bug. If you setup email recovery and then click ‘resend’, wait until it’s done and click ‘resend’ again, two email methods now show up. Re-freshing the page fixes it. I tried to do `refreshAccount` before `loadRecoveryMethod` but it doesn’t help.
![rec-methods](https://user-images.githubusercontent.com/24921205/79398204-e843b600-7f34-11ea-9470-fde48ab7587c.png)

Reply:
https://github.com/nearprotocol/near-wallet/blob/78fb7f0366af88be074d3c331fb9fb1833ec5a9d/src/hooks/recoveryMethods.js#L16

looks like this check results in data being cached and stale